sdk/fmt.sh

```diff
@@ -51,7 +51,7 @@ check_diff() {
   # "${@:3}" command
   changed_files=$(git diff --name-only --diff-filter=ACMRT "$1" | grep $2 | grep -E -v '^canton(-3x)?/' || [[ $? == 1 ]])
   if [[ -n "$changed_files" ]]; then
-    run "${@:3}" ${changed_files[@]:-}
```

Comment:
Extracted here https://github.com/digital-asset/daml/pull/19237

Reply:
> I like this change but in a small test I did the prefix removal only applied to the first file, so:
> 
> * As this is an inf change it may be better to put it in a dedicated PR (@garyverhaegen-da may have an opinion on this)

I really don't - I don't use the pre-commit. My only opinion on the whole thing is it should be opt-in rather than opt-out.